WordPress/gutenberg
# Add fallback handling for plugins or themes using the Customizer

## Description
This change introduces a new function, `gutenberg_site_requires_customizer()`, which looks up if any plugins, or theme features, have added support for the Customizer.

It then utilizes that check before letting Full Site Editing filter away the Customizer access points. It is worth noting that the Customizer is not deactivated even without this patch, but it is important to maintain a canonical entry-point for the user when any plugin, or theme feature, relies on it.

This would fix #35874

## How has this been tested?
1. Activate the Twenty Twenty One theme
2. Install the WooCommerce plugin (no setup required, it just needs to be active)
3. You should now see the Appearance > Customizer menu items
4. Switch to the TT1-Blocks theme
5. The Customizer menu is now missing, but a custom "Customize WooCommerce" link is added instead (this is WooCommerce working around the flaw as described in #35874)
6. Apply this patch
7. The Appearance > Customizer link should now have returned
8. De-activate WooCommerce
9. The Appearance > Customizer link should disapear as no more plugins or theme features rely on it.

## Types of changes
Bug fix (non-breaking change which fixes an issue)

## Checklist:
